ethers-ens/src/main/kotlin/io/ethers/ens/EnsResolver.kt

```diff
@@ -0,0 +1,442 @@
+package io.ethers.ens
+
+import io.ethers.abi.AbiCodec
+import io.ethers.abi.AbiType
+import io.ethers.core.FastHex
+import io.ethers.core.Jackson
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.core.types.Bytes
+import io.ethers.core.types.CallRequest
+import io.ethers.logger.err
+import io.ethers.logger.getLogger
+import io.ethers.logger.wrn
+import io.ethers.providers.Provider
+import io.ethers.providers.types.RpcResponse
+import io.github.adraffy.ens.InvalidLabelException
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+
+class EnsResolver(
+    private val provider: Provider,
+    private val registryAddress: Address,
+    private val ccipLookupLimit: Int = 4,
+    private val client: OkHttpClient = OkHttpClient(),
+) {
+    private val LOG = getLogger()
+    constructor(provider: Provider) : this(provider, getRegistryAddressOrThrow(provider.chainId))
+
+    data class EnsGatewayRequestDTO(val data: Bytes)
+
+    /**
+     * Resolve ens name to Address. On error return [RpcResponse] as error.
+     */
+    fun resolveName(ensName: String): CompletableFuture<RpcResponse<Address>> = CompletableFuture.supplyAsync {
```

Comment:
Would `resolveAddress` perhaps be a better name for this function? Especially if we add `resolveAvatar`/`resolveField` in the future.

Reply:
I think that `resolveName` is OK, because `resolveAddress` may be misunderstood as a reverse operation `lookupAddress`. 
All these operations resolve something regarding ENS name. `resolveField` resolves a field of an ENS name, `resolveAvatar` resolves an avatar of an ENS name.